internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,243 @@
+		return nil, err
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		deployedWkld:       []string{},
+		wkldDeployedToEnvs: make(map[string][]string),
+		prompt:             prompt.New(),
+		store:              store,
+		ws:                 ws,
+		deployStore:        deployStore,
+	}
+	opts.envVersionGetter = func(envName string) (versionGetter, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	//TODO(varun359): Get build information from the manifest and task definition for workloads
+
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		return o.validateEnvName()
+	}
+
+	if len(o.wkldDeployedToEnvs[o.wkldName]) == 1 {
+		log.Infof("Only one environment found, defaulting to: %s\n", color.HighlightUserInput(o.wkldDeployedToEnvs[o.wkldName][0]))
+		o.envName = o.wkldDeployedToEnvs[o.wkldName][0]
+		return nil
+	}
+	selectedEnvName, err := o.prompt.SelectOne("Select an environment in which you want to test", "", o.wkldDeployedToEnvs[o.wkldName], prompt.WithFinalMessage("Environment:"))
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = selectedEnvName
+	return nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionGetter
+
+	checker, err := o.envVersionGetter(envName)
+	if err != nil {
+		return false, err
+	}
+
+	currVersion, err := checker.Version()
+	if err != nil {
+		return false, fmt.Errorf("get environment %q version: %w", envName, err)
+	}
+	if currVersion == version.EnvTemplateBootstrap {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (o *localRunOpts) validateEnvName() error {
+	envs, err := o.deployStore.ListEnvironmentsDeployedTo(o.appName, o.wkldName)
+	if err != nil {
+		return fmt.Errorf("list deployed environments for application %s: %w", o.appName, err)
+	}
+	isDeployed, err := o.isEnvironmentDeployed(o.envName)
+	if err != nil {
+		return err
+	}
+	if !isDeployed {
+		return fmt.Errorf(`cannot use an environment which is not deployed Please run "copilot env deploy, --name %s" to deploy the environment first`, o.envName)
+	}
```

Comment:
```suggestion
```

We might not need to check this anymore, because if it is "bootstrap", then the workload wouldn't have been able to deployed to the environment. Is there any other concern that you've thought of that I've missed?